src/bridges/nft-basic/NftVault.sol

```diff
@@ -0,0 +1,130 @@
+ * @dev This bridge demonstrates basic functionality for an NFT bridge. This may be extended to support more features.
+ */
+contract NftVault is BridgeBase {
+    struct NftAsset {
+        address collection;
+        uint256 id;
+    }
+
+    mapping(uint256 => NftAsset) public tokens;
+    AddressRegistry public immutable REGISTRY;
+
+    event NftDeposit(uint256 indexed virtualAssetId, address indexed collection, uint256 indexed tokenId);
+    event NftWithdraw(uint256 indexed virtualAssetId, address indexed collection, uint256 indexed tokenId);
+
+    /**
+     * @notice Set the addresses of RollupProcessor.sol and AddressRegistry.sol
+     * @param _rollupProcessor Address of the RollupProcessor.sol
+     * @param _registry Address of the AddressRegistry.sol
+     */
+    constructor(address _rollupProcessor, address _registry) BridgeBase(_rollupProcessor) {
+        REGISTRY = AddressRegistry(_registry);
+    }
+
+    /**
+     * @notice Function for the first step of a NFT deposit, or a NFT withdrawal.
+     * @dev This method can only be called from the RollupProcessor.sol. The first step of the
+     * deposit flow returns a virutal asset note that will represent the NFT on Aztec. After the
+     * virutal asset note is received on Aztec, the user calls matchDeposit which deposits the NFT
+     * into Aztec and matches it with the virtual asset. When the virutal asset is sent to this function
+     * it is burned and the NFT is sent to the user (withdraw).
+     *
+     * @param _inputAssetA - ETH (Deposit) or VIRTUAL (Withdrawal)
+     * @param _outputAssetA - VIRTUAL (Deposit) or 0 ETH (Withdrawal)
+     * @param _totalInputValue - must be 1 wei (Deposit) or 1 VIRTUAL (Withdrawal)
+     * @param _auxData - corresponds to the Ethereum address id in the AddressRegistry.sol for withdrawals
+     * @return outputValueA - 1 VIRTUAL asset (Deposit) or 0 ETH (Withdrawal)
+     *
+     */
+
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override (
+            // uint64 _auxData,
+            // address _rollupBeneficiary
+            BridgeBase
+        )
+        onlyRollup
+        returns (uint256 outputValueA, uint256 outputValueB, bool isAsync)
+    {
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidInputA();
+        if (
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidOutputA();
+        if (
```

Comment:
Both cases want `_totalInputValue == 1` so you could perform that check before going into the if-branches to have less deep branches.